client/src/api/datasetCollections.ts

```diff
@@ -19,15 +24,19 @@ export async function fetchCollectionDetails(params: { id: string }): Promise<HD
  * @param params.id The ID of the collection (HDCA) to fetch.
  */
 export async function fetchCollectionSummary(params: { id: string }): Promise<HDCASummary> {
-    const { data } = await getCollectionDetails({ id: params.id, view: "collection" });
+    const { data, error } = await client.GET("/api/dataset_collections/{id}", {
+        params: {
+            path: params,
+            query: { view: "collection" },
+        },
+    });
+
+    if (error) {
+        rethrowSimple(error);
+    }
```

Comment:
Could we instead register a middleware like we did with the old fetcher ?

Reply:
Yes, I have considered it: https://github.com/galaxyproject/galaxy/pull/18532/files#diff-a29bb8e3afc7a891c203b6960df3ff89149f8ef3e4e80658d8a9e35a53d5620aR24

I'll work on that next. I wanted to ask about it in the meeting but couldn't make it. The thing is, I was still unsure because one of the differences between `openapi-fetch` and `openapi-typescript-fetch` is:

> openapi-typescript-fetch throws exceptions for non-OK responses (requiring you to wrap things in try/catch) rather than follow the fetch API spec of not throwing.

Something like this:
### Do not throw
```ts
const { data, error } = await client.GET("/api/libraries");
// here data can be undefined if error is defined. If you try to use it it will fail type validation
// so you are forced to handle the error at "type checking time"
if (error) {
    // handle error, return or throw
}
// data is guaranteed to be defined here
```

So if we add the middleware to rethrow like we were doing, we now probably need to wrap the call in try/catch and, at the same time, we still have to deal with `data` being possibly undefined when `error` is defined like this:

### Throw in middleware
```ts
try{
    const { data, error } = await client.GET("/api/libraries");
    // here data can still be undefined if error is defined
    if (error) {
        // handle error, return or throw
    }
    // data is guaranteed to be defined here
}catch(e){
    // handle error (again?)
}
```

Maybe there is a way to tell the middleware to assume `data` will always be defined as the exception would have been thrown otherwise?

I'm right now a bit on the fence but slightly leaning towards the first option, because it doesn't require you to remember to wrap the call in a try/catch, and on the other hand, it forces you to deal with the error early, otherwise, typescript will complain.

Reply:
If we're committed to handling the optional type and the error in the calling stack that would be a solid improvement for sure 👍. On a quick glance I just saw 

```
const { data, error } = await client.GET("/api/libraries");
// here data can be undefined if error is defined. If you try to use it it will fail type validation
// so you are forced to handle the error at "type checking time"
if (error) {
    rethrowSimple(error)
}
```
in the PR which is why I thought you'd opted for handling errors like we used to, in which case the middleware is less verbose.

Reply:
Yes, those are some of the wrapper functions from "@api/*" that are reused in multiple places. I choose to keep those that were used in more than one place. I'm forced to rethrow there because we don't have the context to handle the error properly and the caller code should be already expecting exceptions.

There are also some situations in which error handling was not considered at all in the calling code or already had a try/catch, in those cases I'm also simply rethrowing to keep the changes minimal. In those cases where I clearly see how to handle the error I'm removing the try/catch and adapting accordingly.